pkg/services/k8s/client/clientset.go

```diff
@@ -48,7 +50,12 @@ type Clientset struct {
 	lock sync.RWMutex
 }
 
-func ProvideClientset() (*Clientset, error) {
+var _ registry.CanBeDisabled = (*Clientset)(nil)
+
+func ProvideClientset(toggles featuremgmt.FeatureToggles) (*Clientset, error) {
+	if !toggles.IsEnabled(featuremgmt.FlagK8s) {
+		return &Clientset{}, nil
```

Comment:
obviously not great... since calls will likely fail with a panic or nil -- but better than breaking all the tests!